app/questionnaire/questionnaire_store_updater.py

```diff
@@ -330,10 +373,8 @@ def update_progress_for_dependant_sections(self) -> None:
                 section_id != self._current_location.section_id
                 or list_item_id != self._current_location.list_item_id
             ):
-                self.update_section_status(
-                    is_complete=False,
-                    section_id=section_id,
-                    list_item_id=list_item_id,
+                self.dependent_sections.add(
+                    DependentSection(section_id, list_item_id, False)
                 )
```

Comment:
Since this is setting the status to `False` we should handle duplicate section dependents. It won't currently handle duplicates as this is setting is_complete to False and the other is setting it to `None`.

I had a think about this and I don't believe, in this scenario, it can ever not be false. If you remove a block for a section, that section would be incomplete, however yes it is possible that another section may become complete etc, but that is not what is being handled here. This is handling status for the section the block was removed from. Therefore, due to the nature of the current call order, we would need to do something like this, which is not the best but avoid duplicates.

```py
				dependent = DependentSection(section_id, list_item_id, None)
                if dependent in self.dependent_sections:
                    self.dependent_sections.remove(dependent)

                self.dependent_sections.add(
                    DependentSection(section_id, list_item_id, False)
                )
```

The other option is to have 1 map in questionnaire store updater to handle both section and block level dependencies. But that would require a bit of refactor, which I don't think is necessary given where we are.

Reply:
I went for option 1 here and just added your suggested code, had to add some more tests.